src/platform/forms-system/src/js/components/AddressViewField.jsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react/jsx-key */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const getField = (formData, possibilities) =>
+  possibilities.reduce((value, field) => {
+    if (value === null && formData[field]) {
+      return formData[field];
+    }
+    return value;
+  }, null);
+
+const addLine = line => line && [line, <br key={line} />];
+
+const AddressViewField = ({ formData }) => {
+  // unchanged address variable names
+  const { country, city, state } = formData;
+
+  // this should cover all current address use cases
+  // street, line2, line3, postalCode = platform address schema
+  // addressLine1, addressLine2, addressLine3 = 526 & HLR
+  // zipCode = multiple forms
+  const street = getField(formData, ['street', 'addressLine1']);
+  const street2 = getField(formData, ['line2', 'street2', 'addressLine2']);
+  const street3 = getField(formData, ['line3', 'street3', 'addressLine3']);
+  const postalCode = getField(formData, ['postalCode', 'zipCode']);
+
+  let postalString = '';
+  if (postalCode) {
+    const lastChunk = postalCode.length > 5 ? `-${postalCode.slice(5)}` : '';
+    postalString = `${postalCode.slice(0, 5)}${lastChunk}`;
+  }
+
+  return (
+    <p>
+      {addLine(street)}
+      {addLine(street2)}
+      {addLine(street3)}
+      {country === 'USA'
+        ? `${city}, ${state} ${postalString}`
+        : `${city}, ${country}`}
+    </p>
+  );
+};
+
+AddressViewField.PropTypes = {
+  formData: PropTypes.shape({
+    country: PropTypes.string.isRequired,
+    city: PropTypes.string.isRequired,
+    state: PropTypes.string,
+
+    street: PropTypes.string,
+    street2: PropTypes.string,
+    street3: PropTypes.string,
+
+    addressLine1: PropTypes.string,
+    addressLine2: PropTypes.string,
+    addressLine3: PropTypes.string,
+
+    postalCode: PropTypes.string,
+    zipCode: PropTypes.string,
```

Comment:
Are any of these required? Conditionally?

Reply:
The `postalCode`/`zipCode` isn't required for non-US countries. I think it's better to let the schema dictate that requirement since newer forms (non-526) still let you enter a state & postal code for non-US countries.

Honestly, I wasn't really sure how to set up these PropTypes without breaking stuff, since the names are all over the place. In fact, I just found that `src/platform/forms/address/helpers.js` uses `addressOne`, `addressTwo` and `addressThree` as names, but since it doesn't use this `AddressViewField` I think we're okay with not covering those variables.